### PR TITLE
feat(packages): add dvanced-statefulset to packages configuration

### PR DESCRIPTION
**Pull request by wuhuizuo**
currently the dockerfile in the repo limit build target platform to linux/amd64.

Ref:
- PingCAP-QE/ci#2628

### CODE REVIEW

File: packages/packages.yaml.tmpl [hunk 1]
```diff
@@ -740,6 +740,35 @@ components:
...
                   path: bin/ng-monitoring-server
+  advanced-statefulset:
+    desc:
+    git:
+      url: https://github.com/pingcap/advanced-statefulset.git
+      ref: '{{ .Git.ref | default "master" }}'
+      sha: '{{ .Git.sha | default "" }}'
+    version: {{ .Release.version }} # segment version.
+    routers: # match once.
+      - description: interpret versions according to semantic version spec.
```
Comment (wuhuizuo) [inline packages/packages.yaml.tmpl:751]: Publish route for v0.5.0 and newer versions.

Comment (wuhuizuo) [inline packages/packages.yaml.tmpl:751]:
```suggestion
      - description: Publish route for v0.5.0 and newer versions.
```

**Comment by wuhuizuo** [comment]
> > **I have already done a preliminary review for you, and I hope to help you do a better job.**
> > **Review:**
> 
> This pull request adds `advanced-statefulset` to the packages configuration in the repository.
> 
> **Key Changes:**
> 
> * Added `advanced-statefulset` to packages configuration in `packages.yaml.tmpl`.
> * Modified `build-package-images.sh.tmpl` to build and push images with the correct tag.
> * Added `Dockerfile` for `advanced-statefulset`.
> 
> **Potential Problems:**
> 
> The pull request seems to be fine and doesn't raise any significant problems. However, some minor recommendations should be considered:
> 
> * It is advisable to use the full SHA instead of truncating it to 7 characters.
> * The Dockerfile in the repo limits building target platform to linux/amd64, so it should be taken into account whether this is intentional or not.
> 
> **Fixing Suggestions:**
> 
> * Use the full SHA instead of truncating it to 7 characters.
> * Consider adding support for other platforms in the Dockerfile if needed.
> 
> Overall, the pull request looks good, and the changes seem to be useful. Therefore, I recommend merging it into the master branch.

@csuzhangxc Please check this:
> The Dockerfile in the repo limits building target platform to linux/amd64, so it should be taken into account whether this is intentional or not.

**Comment by csuzhangxc** [comment]
it's better to support both AMD64 and ARM, I added a `TARGETARCH` arg in the Dockerfile.

ref https://github.com/pingcap/advanced-statefulset/pull/101

**Pull request merged**